brain-marks/Categories/Views/CategoryList.swift

```diff
@@ -48,7 +48,55 @@ struct CategoryList: View {
             viewModel.getCategories()
         }
         .accentColor(.black)
-        .sheet(isPresented:$showAddURLView) {
+         .sheet(isPresented:$showAddURLView) {
+                    AddURLView(categories: viewModel.categories)
+                }
+    }
+}
+
+struct NewCategorySheetView: View {
+    @Environment(\.presentationMode) var presentationMode
+    @State var newCategory = ""
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                TextField("Enter name of new category", text: $newCategory)
+                    .foregroundColor(.gray)
+                
+                
+                Spacer()
+                
+                HStack(spacing: 25) {
+                    Button {
+                        presentationMode.wrappedValue.dismiss()
+                    } label: {
+                        Text("Cancel")
+                            .frame(width: 150, height: 50)
+                            .foregroundColor(.white)
+                            .background(Color(UIColor(named: "twitter")!))
+                            .font(.system(size: 20, weight: .semibold, design: .default))
+                            .cornerRadius(10)
+                    }
+                    
+                    Button {
+                        presentationMode.wrappedValue.dismiss()
+                        DataStoreManger.shared.createCategory(
+                            category: AWSCategory(name: newCategory,
+                                                  imageName: "swift"))
+                    } label: {
+                        Text("Create")
+                            .frame(width: 150, height: 50)
+                            .foregroundColor(.white)
+                            .background(Color(UIColor(named: "twitter")!))
+                            .font(.system(size: 20, weight: .semibold, design: .default))
+                            .cornerRadius(10)
+                    }
+                }
+                .padding(20)
+            }
+            .navigationBarTitle("New Category")
+            .sheet(isPresented:$showAddURLView) {
```

Comment:
All of this shouldn't be here. There is a separate file called NewCategorySheetView. It's located at: brain-marks/Categories/Views

Do you has this file?